### DISCUSSION TITLE
Trying to make a countdown timer for video recording, but the callback is fired only once.

### DISCUSSION THREAD

**Discussion by ykiran** [mateusfccp/pausable_timer#59] [Q&A] [answered]
I tried the same with the inbuilt Timer.Periodic(), but pausing and resuming was so erratic; I had a time limit of 10 seconds, but pausing and resuming multiple times causes the total recording time to go over that limit.
I stumbled across this library from an SO answer.

When I replaced Timer.Periodic with PausableTimer, I see that it is fired only once.

Here is the sample code:

```
 int _start= 10;

 void startTimer() {    
    _timer = PausableTimer(
      Duration(seconds: 1),
      () {
        if (_start <= 0) {         
          recordVideo();
          setState(() {
            _isRecording = true;
          });
        } else {
          setState(() {
            _start = _start - 1;            
          });          
        }
      },
    )..start();
  }

```
This method will be called when the recording button is pressed.
**recordVideo()** is what I call to either start recording (along with **startTime()**) or end recording. 

I know you're busy and haven't touched Dart in years, but can you help me understand what is that I'm doing wrong here and why is it being called only once?

**Comment by llucax** [comment, marked as answer]
Hi @ykiran, I'm sorry but just from that code snippet is very hard to know what's going on but if what you want is a periodic timer, you need to restart the timer yourself after it is fired. Please have a look at the [countdown example](https://github.com/llucax/pausable_timer/blob/main/example/countdown.dart).

**Reply by llucax** [reply]
Marking as answered, I assume from the lack of replies that this was solved.
